src/internal/ppsutil/util.go

```diff
@@ -117,6 +117,9 @@ func GetLimitsResourceList(limits *pps.ResourceSpec) (*v1.ResourceList, error) {
 func GetPipelineInfoAllowIncomplete(pachClient *client.APIClient, ptr *pps.StoredPipelineInfo) (*pps.PipelineInfo, error) {
 	result := &pps.PipelineInfo{}
 	buf := bytes.Buffer{}
+	// ensure we are authorized to read the pipeline's spec commit, but don't propagate that back out
+	pachClient = pachClient.WithCtx(pachClient.Ctx())
+	pachClient.SetAuthToken(ptr.AuthToken)
```

Comment:
previously, the (singular) spec repo gave everyone read permissions
While this grants more permissions, it seems like limiting the scope to this function call eliminates any issues (especially considering the caller has the pipeline info with the auth token, too)

Reply:
This is a fine solution for now. AFAICT these methods are used in pachd and the worker, but we really want two different implementations - the worker version should accept a client with the PPS token, but the PPS version shouldn't. We can clean that up in the future.

Reply:
I'm not sure I fully understand the distinction you're making, but sounds good